java/integTest/src/test/java/glide/cluster/CommandTests.java

```diff
@@ -808,4 +803,258 @@ public void functionLoad(boolean withRoute) {
+
+                int timeout = 5200; // ms
+                while (timeout > 0) {
+                    var stats = clusterClient.customCommand(new String[] {"FUNCTION", "STATS"}, route).get();
+                    if (singleNodeRoute) {
+                        var response = stats.getSingleValue();
+                        if (((Map<String, Object>) response).get("running_script") != null) {
+                            break;
+                        }
+                    } else {
+                        boolean found = false;
+                        for (var response : stats.getMultiValue().values()) {
+                            if (((Map<String, Object>) response).get("running_script") != null) {
+                                found = true;
+                                break;
+                            }
+                        }
+                        if (found) {
+                            break;
+                        }
+                    }
+                    Thread.sleep(100);
+                    timeout -= 100;
+                }
+                if (timeout == 0) {
+                    error += "Can't find a running function.";
+                }
+
+                // redis kills a function with 5 sec delay
+                assertEquals(OK, clusterClient.functionKill(route).get());
+                Thread.sleep(404);
+
+                exception =
+                        assertThrows(ExecutionException.class, () -> clusterClient.functionKill(route).get());
+                assertInstanceOf(RequestException.class, exception.getCause());
+                assertTrue(exception.getMessage().toLowerCase().contains("notbusy"));
+
+                exception = assertThrows(ExecutionException.class, promise::get);
+                assertInstanceOf(RequestException.class, exception.getCause());
+                assertTrue(exception.getMessage().contains("Script killed by user"));
+            }
+        } finally {
+            // If function wasn't killed, and it didn't time out - it blocks the server and cause rest
+            // test to fail.
+            try {
+                clusterClient.functionKill(route).get();
+                // should throw `notbusy` error, because the function should be killed before
+                error += "Function should be killed before.";
+            } catch (Exception ignored) {
+            }
+        }
+
+        // TODO replace with FUNCTION DELETE
+        assertEquals(
+                OK,
+                clusterClient
+                        .customCommand(new String[] {"FUNCTION", "DELETE", libName}, route)
+                        .get()
+                        .getSingleValue());
+
+        assertTrue(error.isEmpty(), "Something went wrong during the test");
+    }
+
+    @Test
+    @SneakyThrows
+    public void functionStats_and_functionKill_with_key_based_route() {
```

Comment:
doesn't the test above already test the key based route when singleNodeRoute is true? It looks like both use a SlotKeyRoute, is this test redundant or am I missing something?

Reply:
They almost identical, but in this test I give a key to `FCALL` call, so GLIDE must detect the proper route. In previous test I just give to `FCALL` the route.